apps/dashboard/src/components/workflow-editor/steps/edit-step-sidebar.tsx

```diff
@@ -10,45 +9,25 @@ import {
   SheetPortal,
   SheetTitle,
 } from '@/components/primitives/sheet';
-import { useFetchWorkflow } from '@/hooks/use-fetch-workflow';
 import { StepEditor } from '@/components/workflow-editor/steps/step-editor';
-import { useFetchStep } from '@/hooks/use-fetch-step';
 import { VisuallyHidden } from '@/components/primitives/visually-hidden';
 import { PageMeta } from '@/components/page-meta';
-import { getStepBase62Id } from '@/utils/step';
-import { EXCLUDED_EDITOR_TYPES } from '@/utils/constants';
+import { StepSkeleton } from './step-skeleton';
+import { StepEditorProvider } from './step-editor-provider';
+import { useStepEditorContext } from './hooks';
+import { useWorkflowEditorContext } from '../hooks';
 
 const transitionSetting = { ease: [0.29, 0.83, 0.57, 0.99], duration: 0.4 };
 
-export const EditStepSidebar = () => {
-  const { workflowSlug = '', stepSlug = '' } = useParams<{ workflowSlug: string; stepSlug: string }>();
+const EditStepSidebarInternal = () => {
   const navigate = useNavigate();
-
-  const { workflow } = useFetchWorkflow({
-    workflowSlug,
-  });
-
-  const { step } = useFetchStep({ workflowSlug, stepSlug });
-  const stepType = useMemo(
-    () => workflow?.steps.find((el) => getStepBase62Id(el.slug) === getStepBase62Id(stepSlug))?.type,
-    [stepSlug, workflow]
-  );
```

Comment:
Moved to the `StepEditorProvider` component which shares the step across the components.
I decided to do that because, in a few places, we were fetching the step with the `useFetchStep` hook, which on mount fetches the data from API, triggering the loading skeleton to be shown, and it was infinite because the skeleton was in the parent.